### PR TITLE
fix: add missing change event

### LINKED ISSUE #109: Does not fire `change` event on change
Should be aligned with the rest of elements.

### PR DESCRIPTION
**Pull request by web-padawan**
Fixes #109

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-time-picker/110)

### CODE REVIEW

**Comment by web-padawan** [comment]
@tomivirkki fixed, should now reset flag properly. Added a test emulating that case.

File: src/vaadin-time-picker.html [hunk 1]
```diff
@@ -377,6 +377,24 @@
           // thus querySelector in textField.focusElement raises an undefined exception on validate
           this.__dropdownElement.addEventListener('value-changed', e => this.__onInputChange(e));
           this.__inputElement.addEventListener('keydown', this.__onKeyDown.bind(this));
+
+          this.__dropdownElement.addEventListener('change', e => {
+            if (e.composedPath()[0] === this.__inputElement) {
+              if (e.detail.sourceEvent.__fromClearButton) {
+                // Clear button clicked. Delay change event until actual value change
+                this.__pendingChange = true;
```
Comment (tomivirkki) [inline src/vaadin-time-picker.html]: Hmm...I can comment out this row but still the whole test suite passes (even "should fire change on clear button click").

Also the same thing with the lines below (with `this.__pendingChange = false;`).

Comment (web-padawan) [reply src/vaadin-time-picker.html]: Good catch. Updated and simplified logic to listen for 'change' event from `vaadin-combo-box-light`.
Looks like it forwards event in all the cases so we only need to filter out duplicate events from text-field.

**Comment by web-padawan** [comment]
> One more thing: if the resolution is low enough for the time picker not to open an overlay, changing the value with up/down keys should also dispatch a change event (align with, `<input type="number">` and `<vaadin-number-field>`)

Done.

**Pull request merged**